src/scan_validator.cpp

```diff
@@ -0,0 +1,105 @@
+/*
+ * (c) Copyright 2015-2016 6 River Systems, all rights reserved.
+ *
+ * This is proprietary software, unauthorized distribution is not permitted.
+ *
+ * scan_validator.h
+ *
+ * created: 09/12/2017
+ */
+
+#include <math.h>
+#include <ros/ros.h>
+#include <srslib_framework/math/BasicMath.hpp>
+#include "laser_filters/scan_validator.h"
+
+laser_filters::ScanValidator::ScanValidator() :
+  has_checked_laser_config_(false)
+{
+
+}
+
+laser_filters::ScanValidator::~ScanValidator()
+{
+
+}
+
+bool laser_filters::ScanValidator::configure()
+{
+  // Setup default values
+  getParam("violation_percentage", violation_percentage_);
+  getParam("contour_tolerance", contour_tolerance_);
+  getParam("frame_id", frame_id_);
+  getParam("angle_min", angle_min_);
+  getParam("angle_max", angle_max_);
+  getParam("angle_increment", angle_increment_);
+  getParam("range_min", range_min_);
+  getParam("range_max", range_max_);
+  getParam("contour", contour_);
+
+  return true;
+}
+
+bool laser_filters::ScanValidator::update(
+    const sensor_msgs::LaserScan& input_scan,
+    sensor_msgs::LaserScan &output_scan)
+{
+  // Copy input_scan data
+  output_scan = input_scan;
```

Comment:
Should probably only copy the data to the output when the input is valid (i.e. all true cases.).